src/persistence/keyvalue/NoStorageMapKeyValueStore.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { KeyValueHandler } from "./KeyValueHandler";
+
+export class NoStorageMapKeyValueStore implements KeyValueHandler {
+    protected data = new Map<string, any>();
+
+    constructor() {}
+
+    private buildKeyString(context: string, contextKey: string): string {
+        return `${context}$$${contextKey}`;
+    }
+
+    getContextKey(context: string, contextKey: string, defaultValue: any): any | undefined {
+        const value = this.data.get(this.buildKeyString(context, contextKey));
+        if (value === undefined) {
+            return defaultValue;
+        }
+        return value;
```

Comment:
Is `null` an allowed value? If not, you could just do
```suggestion
        return this.data.get(this.buildKeyString(context, contextKey)) ?? defaultValue;
```

Reply:
Formally I would like to allow null, so the "undefined" check is there to use the default vaue if "the key is not set" aka undefined (ok I could also do a has before, but undefined check should be equivalentm right?